diff --git a/src/extension/internal/pdfinput/svg-builder.cpp b/src/extension/internal/pdfinput/svg-builder.cpp
index 158c5ae068..db33868805 100644
--- a/src/extension/internal/pdfinput/svg-builder.cpp
+++ b/src/extension/internal/pdfinput/svg-builder.cpp
@@ -111,7 +111,7 @@ SvgBuilder::~SvgBuilder() {
 void SvgBuilder::_init() {
     _font_style = NULL;
     _current_font = NULL;
-    _font_specification = NULL;
+    const char* _font_specification = NULL;
     _font_scaling = 1;
     _need_font_update = true;
     _in_text_object = false;
